py/grammar.h

```diff
@@ -239,15 +253,27 @@ DEF_RULE(term_op, nc, or(4), tok(OP_STAR), tok(OP_SLASH), tok(OP_PERCENT), tok(O
 DEF_RULE(factor, nc, or(2), rule(factor_2), rule(power))
 DEF_RULE(factor_2, c(factor_2), and(2), rule(factor_op), rule(factor))
 DEF_RULE(factor_op, nc, or(3), tok(OP_PLUS), tok(OP_MINUS), tok(OP_TILDE))
+#if MICROPY_ASYNC_SYNTAX
+DEF_RULE(power, c(power), and(2), rule(await_or_atom_expr), opt_rule(power_dbl_star))
+DEF_RULE(await_or_atom_expr, nc, or(2), rule(await_expr), rule(atom_expr))
+DEF_RULE(await_expr, c(await_expr), and(3), tok(KW_AWAIT), rule(atom), opt_rule(atom_expr_trailers))
+DEF_RULE(atom_expr, c(atom_expr), and(2), rule(atom), opt_rule(atom_expr_trailers))
+DEF_RULE(atom_expr_trailers, c(atom_expr_trailers), one_or_more, rule(trailer))
+#else
 DEF_RULE(power, c(power), and(3), rule(atom), opt_rule(power_trailers), opt_rule(power_dbl_star))
 DEF_RULE(power_trailers, c(power_trailers), one_or_more, rule(trailer))
+#endif
 DEF_RULE(power_dbl_star, nc, ident | and(2), tok(OP_DBL_STAR), rule(factor))
 
-// atom: '(' [yield_expr|testlist_comp] ')' | '[' [testlist_comp] ']' | '{' [dictorsetmaker] '}' | NAME | NUMBER | STRING+ | '...' | 'None' | 'True' | 'False'
+// atom: '(' [yield_expr|testlist_comp] ')' | '[' [testlist_comp] ']' | '{' [dictorsetmaker] '}' | NAME | NUMBER | STRING+ | '...' | 'None' | 'True' | 'False' | await_expr
```

Comment:
This change is not in the CPython grammar specification.  It shouldn't be needed.  I think any "await" keyword is parsed already by the new "power" rule above.